tests/test_completion.py

```diff
@@ -152,7 +153,12 @@ def test_shell_integration(
 
         # Fish starts completion with one tab compared to bash's two tabs.
         if shell == "fish":
-            num_tabs -= 1
+            # Fish will add a space to an unambiguous completion.
+            expected = [x + " " if re.match(r".*=\w+$", x) else x for x in expected]
+
+            # Exactly match token end. See
+            # https://github.com/fish-shell/fish-shell/issues/6928
+            expected = [re.escape(x) + "$" for x in expected]
```

Comment:
can we do this unconditionally?

Reply:
For bash as well? I don't think the current expect script for bash supports doing this way.

Reply:
Yes, I am trying to minimize the things that are making fish special. your suggesting to test against eol makes sense for bash as well, with a bit of luck it will just pass if you do it there and add the -re to the expect match.